src/ProjectTemplates/BlazorTemplates.Tests/Infrastructure/TemplateTests.props.in

```diff
@@ -29,11 +29,14 @@
       IsImplicitlyDefined="true" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(UsingMicrosoftNETSdkWeb)' == 'true' OR '$(RazorSdkCurrentVersionProps)' != ''">
-    <!--
-      Use the Razor SDK as a package reference. The version of the .NET Core SDK we build with often contains a version of the Razor SDK
-      several versions older than latest. To avoid a cyclical dependency, this package reference is added to override the bundled version.
-    -->
-    <PackageReference Include="Microsoft.NET.Sdk.Razor" PrivateAssets="All" Version="${MicrosoftNETSdkRazorPackageVersion}" />
-  </ItemGroup>
+  <!--
+    Use the Razor SDK as a project reference. The version of the .NET Core SDK we build with often contains a version of the Razor SDK
+    several versions older than latest. To avoid a cyclical dependency, this package reference is added to override the bundled version.
+    Since this is a project reference, we must explicitly import the props file and also specify the output location of the SDK directory.
+  -->
+  <Import Project="${RepoRoot}src\Razor\Microsoft.NET.Sdk.Razor\src\build\netstandard2.0\Microsoft.NET.Sdk.Razor.props" Condition="'$(UsingMicrosoftNETSdkWeb)' == 'true' OR '$(RazorSdkCurrentVersionProps)' != ''" />
+
+  <PropertyGroup>
+    <RazorSdkDirectoryRoot>${ArtifactsBinDir}Microsoft.NET.Sdk.Razor\${Configuration}\sdk-output\</RazorSdkDirectoryRoot>
+  </PropertyGroup>
```

Comment:
Again, move this repeated logic into a single file under eng/

Reply:
Turns out this is a little more nuanced and I'm inclined to keep as is. 

The issue here is that while the SDK was built under the Release configuration, the tests invoke dotnet build or dotnet msbuild /t:Build in the Debug configuration. If I were to simply import targets from under eng/targets, it would mean that we need additional override logic to override the configuration property when running tests. I tried it out and it looks rather ugly. Instead, using the paths burned in during test compilation which is run under the Release configuration is a rather elegant solution IMO.

Reply:
The logic in eng/targets/CSharp.Common.props also uses `$(Configuration)` as an input. Why must this new `<Import/>` and `<PropertyGroup/>` be repeated six times? The logic _will_ need to change and having it spread out like this will cause problems in the future.

Reply:
As I said, this is rather tricky because the tests are compiled under the release configuration but the tests assets are built under the Debug configuration. I'll file a followup issue since consolidating the logic in one place might not be trivial. I have an idea to try but it'll take a little bit of trial and error.